### PR TITLE
Fix #20006 - Add Address Details and View on Explorer to Global Menu

### LINKED ISSUE #20006: Duplicate account block explorer and account details links within Global Menu
## Background
There've been numerous reports that users can't find the "block explorer" and "account detail" links. These have been moved under each Account within its submenu item. We should retain those links but duplicate it within the Global Menu.

## Designs
[Figma](https://www.figma.com/file/LDlI7yjKBNIY60MtQv9QS0/EVM-only-Multichain-v1?type=design&node-id=2963-180730&mode=design&t=cXbvaVoHRGUyLcYj-0)

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

* Fixes #20006

Adds the "Address Details" and "View on Explorer" items from the account list item menu to the new global menu

## ToDo

- [ ]  Move the newly shared menu items to new components and new directory
- [ ] Update tests to ensure they mock functions are called with the correct data

## Screenshots/Screencaps

SCR-20230713-odxh

## Manual Testing Steps

1.  Open global menu
2. Ensure "Account Details" provides the correct account
3. Ensure that "View on Explorer" opens the correct link

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by DDDDDanica — DISMISSED** [review]

File: ui/components/multichain/global-menu/global-menu.js [hunk 1]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+  const { blockExplorerUrl } = rpcPrefs;
+  const blockExplorerUrlSubTitle = getURLHostName(blockExplorerUrl);
+  const blockExplorerLinkText = useSelector(getBlockExplorerLinkText);
+  const openBlockExplorer = () => {
+    trackEvent({
+      event: MetaMetricsEventName.ExternalLinkClicked,
+      category: MetaMetricsEventCategory.Navigation,
+      properties: {
+        link_type: MetaMetricsEventLinkType.AccountTracker,
+        location: 'Account Options',
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:108]: Change this to `Global Menu`

File: ui/components/multichain/global-menu/global-menu.js [hunk 2]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
     <Menu anchorElement={anchorElement} onHide={closeMenu}>
+      <MenuItem
+        onClick={() => {
+          console.log('Current address is: ', currentAddress);
+          dispatch(setAccountDetailsAddress(currentAddress));
+          trackEvent({
+            event: MetaMetricsEventName.NavAccountDetailsOpened,
+            category: MetaMetricsEventCategory.Navigation,
+            properties: {
+              location: 'Account Options',
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:133]: Change this to `Global Menu`

File: ui/components/multichain/global-menu/global-menu.js [hunk 3]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+
+  const routeToAddBlockExplorerUrl = () => {
+    history.push(`${NETWORKS_ROUTE}#blockExplorerUrl`);
+  };
+
   return (
     <Menu anchorElement={anchorElement} onHide={closeMenu}>
+      <MenuItem
+        onClick={() => {
+          console.log('Current address is: ', currentAddress);
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:127]: Remove `console.log`

File: ui/components/multichain/global-menu/global-menu.js [hunk 4]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+        onClick={() => {
+          blockExplorerLinkText.firstPart === 'addBlockExplorer'
+            ? routeToAddBlockExplorerUrl()
+            : openBlockExplorer();
+
+          trackEvent({
+            event: MetaMetricsEventName.BlockExplorerLinkClicked,
+            category: MetaMetricsEventCategory.Accounts,
+            properties: {
+              location: 'Account Options',
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:153]: Change to `Global menu`

File: ui/components/multichain/global-menu/global-menu.js (hunk 2 above)
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:133]:
```suggestion
              location: 'Global Menu',
```

File: ui/components/multichain/global-menu/global-menu.js (hunk 3 above)
Comment (darkwing) [reply ui/components/multichain/global-menu/global-menu.js:127]:
```suggestion
```

File: ui/components/multichain/global-menu/global-menu.js [hunk 5]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+            event: MetaMetricsEventName.NavAccountDetailsOpened,
+            category: MetaMetricsEventCategory.Navigation,
+            properties: {
+              location: 'Account Options',
+            },
+          });
+          closeMenu?.();
+        }}
+        iconName={IconName.ScanBarcode}
+        data-testid="account-list-menu-details"
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:139]:
```suggestion
        data-testid="global-menu-details"
```

File: ui/components/multichain/global-menu/global-menu.js (hunk 4 above)
Comment (darkwing) [reply ui/components/multichain/global-menu/global-menu.js:153]:
```suggestion
              location: 'Global Menu',
```

File: ui/components/multichain/global-menu/global-menu.js [hunk 6]
```diff
@@ -83,8 +92,75 @@ export const GlobalMenu = ({ closeMenu, anchorElement }) => {
...
+            category: MetaMetricsEventCategory.Accounts,
+            properties: {
+              location: 'Account Options',
+              chain_id: chainId,
+            },
+          });
+        }}
+        subtitle={blockExplorerUrlSubTitle || null}
+        iconName={IconName.Export}
+        data-testid="account-list-menu-open-explorer"
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.js:160]:
```suggestion
        data-testid="global-menu-open-explorer"
```

File: ui/components/multichain/global-menu/global-menu.test.js [hunk 7]
```diff
@@ -18,11 +18,34 @@ const render = (metamaskStateChanges = {}) => {
...
 
 describe('AccountListItem', () => {
+  it('renders "View on explorer"', () => {
+    global.platform = { openTab: jest.fn() };
+
+    const { getByText, getByTestId } = render();
+    expect(getByText('View on explorer')).toBeInTheDocument();
+
+    const openExplorerTabSpy = jest.spyOn(global.platform, 'openTab');
+    fireEvent.click(getByTestId('account-list-menu-open-explorer'));
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.test.js:35]:
```suggestion
    fireEvent.click(getByTestId('global-menu-open-explorer'));
```

File: ui/components/multichain/global-menu/global-menu.test.js [hunk 8]
```diff
@@ -18,11 +18,34 @@ const render = (metamaskStateChanges = {}) => {
...
+    expect(openExplorerTabSpy).toHaveBeenCalled();
+  });
+
+  it('opens the Account Details modal with the correct address', () => {
+    global.platform = { openTab: jest.fn() };
+
+    const { getByText, getByTestId } = render();
+    expect(getByText('Account details')).toBeInTheDocument();
+
+    fireEvent.click(getByTestId('account-list-menu-details'));
```
Comment (darkwing) [inline ui/components/multichain/global-menu/global-menu.test.js:45]:
```suggestion
    fireEvent.click(getByTestId('global-menu-details'));
```

**Review by NidhiKJha — APPROVED** [review]
LGTM 🚀

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**